src/network-analytics/azext_network_analytics/aaz/latest/network_analytics/data_product/_ingest.py

```diff
@@ -0,0 +1,168 @@
+            required=True,
+        )
+        _args_schema.source = AAZStrArg(
+            options=["--srcdir"],
+            arg_group="Body",
+            help="Source directory path.",
+            required=True,
+        )
+        return cls._args_schema
+    
+    def _execute_operations(self):
+        self.pre_operations()
+        self.DataProductsIngest(ctx=self.ctx)()
+        self.post_operations()
+
+    @register_callback
+    def pre_operations(self):
+        pass
+
+    @register_callback
+    def post_operations(self):
+        pass
+
+    def _output(self, *args, **kwargs):
+        result = self.deserialize_output(self.ctx.vars.instance, client_flatten=True)
+        return result
+    
+    class DataProductsIngest(object):
+        DATA_PRODUCT_ARM_ID = "/subscriptions/{}/resourceGroups/{}/providers/Microsoft.NetworkAnalytics/dataProducts/{}"
+        API_VERSION = "2023-11-15"
+        KEYVAULT_NAME = "input-storage-sas"
+        KEYVAULT_URI = "https://aoi-{}-kv.vault.azure.net/"
+
+        def __init__(self, ctx):
+            self.ctx = ctx
+            self.subscription_id = ctx.subscription_id
+            self.resource_group = ctx.args.resource_group
+            self.data_product_name = ctx.args.data_product_name
+            self.data_type = ctx.args.data_type
+            self.source = ctx.args.source
+
+            self.resources_client = get_mgmt_service_client(self.ctx.cli_ctx, ResourceType.MGMT_RESOURCE_RESOURCES, subscription_id=self.subscription_id).resources
+            self.container_client = get_sdk(self.ctx.cli_ctx, ResourceType.DATA_STORAGE_BLOB, '_container_client#ContainerClient')
+
+        def __call__(self, *args, **kwargs):
+            data_product = self.get_data_product()
+            secret = self.get_key_vault_secret(data_product)
+            self.upload_file(secret)
+
+        def get_data_product(self):
+            arm_id = self.DATA_PRODUCT_ARM_ID.format(self.subscription_id, self.resource_group, self.data_product_name)
+            resource = self.resources_client.get_by_id(arm_id, self.API_VERSION)
+            return resource
+        
+        def get_keyvault_url(self, data_product):
+            consumption_endpoints = data_product.properties['consumptionEndpoints']
+            ingestion_url = consumption_endpoints['ingestionUrl']
+            unique_id = re.search("https://aoiingestion(.*)\.blob\.core\.windows\.net", ingestion_url).group(1)
+            vault_base_url = self.KEYVAULT_URI.format(unique_id)
+            return vault_base_url
+        
+        def get_key_vault_secret(self, data_product):
+            keyvault_url = self.get_keyvault_url(data_product)
+            command_args = {'vault_base_url': keyvault_url}
+            keyvault_client  = data_plane_azure_keyvault_secret_client(self.ctx.cli_ctx, command_args)
+            try:
+                secret = keyvault_client.get_secret(name=self.KEYVAULT_NAME)
+            except ClientAuthenticationError:
```

Comment:
The key vault exists int he customers subscription, so its entirely possible for them to rename the managed resources resource group, key vault and/or secret. I assume this would break any management of that resource by the data product itself, but it is possible. So we should probably add some error handling that returns a helpful message if the expected resource group, key vault or secret does not exist.